src/main/java/com/google/step/snippet/external/W3SchoolsClient.java

```diff
@@ -49,9 +51,18 @@ public Card search(String w3Link) {
     if (snippets.isEmpty() || snippets.first().getElementsByClass(CODE_CLASS).text().isEmpty()) {
       return null;
     }
-    String title = titles.first().text();
-    String description = descriptions.first().text();
-    String code = snippets.first().getElementsByClass(CODE_CLASS).text();
+    String title =
+        StringEscapeUtils.escapeHtml4(
+            Jsoup.clean(titles.first().text(), Whitelist.basicWithImages()));
+    String description =
+        StringEscapeUtils.escapeHtml4(
+            Jsoup.clean(descriptions.first().text(), Whitelist.basicWithImages()));
+    String code =
+        Jsoup.clean(
+            snippets.first().getElementsByClass(CODE_CLASS).text(), Whitelist.basicWithImages());
+    if (query.contains("html")) {
+      code = StringEscapeUtils.escapeHtml4(code);
+    }
```

Comment:
duplicate code here in two clients, can move to either Client interface or Card class